code/__HELPERS/path.dm

```diff
@@ -173,20 +187,37 @@
 		for(var/scan_direction in list(NORTHEAST, SOUTHEAST, NORTHWEST, SOUTHWEST))
 			diag_scan_spec(current_turf, scan_direction, current_processed_node)
 
-		CHECK_TICK
+		// Stable, we'll just be back later
+		if(TICK_CHECK)
+			return TRUE
+	return TRUE
+
+/**
+ * early_exit() is called when something goes wrong in processing, and we need to halt the pathfinding NOW
+ */
+/datum/pathfind/proc/early_exit()
+	on_finish.Invoke(null)
+	on_finish = null
+	qdel(src)
```

Comment:
Since Destroy calls this, wouldn't this call the callback twice? Would mean that bots would say 'unable to reach destination' twice, for example.

Reply:
well we null out our reference to it, so I'd hope not no